src/components/CertificateViewer.tsx

```diff
@@ -59,13 +59,15 @@ export const CertificateViewer = ({
   and tokenRegistryAddress provided and clean up the context when certificate viewer unmounts
   */
   useEffect(() => {
-    trace("initialise token information context");
-    initialize(tokenRegistryAddress, tokenId);
-    return () => {
-      trace("reseting token information on unmount");
-      resetTokenInformationState();
-      resetCertificateData();
-    };
+    if (tokenRegistryAddress) {
```

Comment:
Looks like there are alot of components depending on the Token Registry address maybe we can put a spinner/loading if it's undefined?

Reply:
Actually, the tokenRegistryAddress can be null when it's not a token. If it exist, it will be shown immediately, its not derived from side effects but from the document directly. This PR just makes it more explicit in the code, just wondering why hasnt it crash in runtime before through...